cockroach-admin/src/cockroach_cli.rs

```diff
@@ -296,4 +403,166 @@ mod tests {
+                    panic!(
+                        "failed to wait for listen addr at \
+                         {listen_addr_path}: {err:?}",
+                    );
+                }
+            };
+
+            Self {
+                _tempdir: tempdir,
+                child,
+                listen_addr,
+                url: format!("postgres://{listen_addr}"),
+            }
+        }
+    }
+
+    // Ensure that if `cockroach init` changes in a future CRDB version bump, we
+    // have a test that will fail to force us to check whether our current
+    // parsing is still valid.
+    #[tokio::test]
+    async fn test_cluster_init_compatibility() {
+        let logctx = dev::test_setup_log("test_cluster_init_compatibility");
+
+        {
+            let db = UninitializedCockroach::start().await;
+
+            // Manually run `cockroach init` to grab the output string and check
+            // that it's exactly what we expect.
+            let mut command = Command::new("cockroach");
+            command.arg("init").arg("--insecure").arg("--url").arg(&db.url);
+            let output = exec_command_logging_output(&mut command).await;
+            assert!(output.status.success());
+            assert_eq!(output.stdout, "Cluster successfully initialized\n");
+
+            // Run it again to confirm the "already initizlied" string.
+            let stderr = exec_command_logging_output(&mut command).await.stderr;
+            assert!(
+                stderr.starts_with(CLUSTER_ALREADY_INIT),
+                "unexpected stderr: {stderr:?}"
+            );
+        }
+
+        // Repeat the above test but using our wrapper.
+        {
+            let db = UninitializedCockroach::start().await;
+            let cli = CockroachCli::new("cockroach".into(), db.listen_addr);
+
+            cli.cluster_init().await.expect("cluster initialized");
+            cli.cluster_init().await.expect("cluster still initialized");
+        }
+
+        logctx.cleanup_successful();
+    }
+
+    #[tokio::test]
+    async fn test_schema_init() {
+        let logctx = dev::test_setup_log("test_schema_init");
+        let db = TestDatabase::new_populate_nothing(&logctx.log).await;
+        let db_url = db.crdb().listen_url().to_string();
+        let url: Url = db_url.parse().expect("valid url");
+        let cockroach_address: SocketAddrV6 = format!(
+            "{}:{}",
+            url.host().expect("url has host"),
+            url.port().expect("url has port")
+        )
+        .parse()
+        .expect("valid SocketAddrV6");
+
+        let cli = CockroachCli::new("cockroach".into(), cockroach_address);
+        cli.schema_init_impl("../schema/crdb/dbinit.sql")
+            .await
+            .expect("initialized schema");
+        cli.schema_init_impl("../schema/crdb/dbinit.sql")
+            .await
+            .expect("initializing schema is idempotent");
+
+        db.terminate().await;
+        logctx.cleanup_successful();
+    }
```

Comment:
Up to you, but might be worth including a test of:

- Cluster Init -> OK
- Schema Init -> OK
- Cluster Init again -> Still OK
- Schema Init again -> Still OK

(this would help validate that the back-to-back calls in `cockroach-admin/src/http_entrypoints.rs` are fine)

If we *really* care, we could insert some data in the schema and check it still exists, but considering the "wiping" is no longer happening, this seems less necessary